src/main/java/it/eng/idsa/businesslogic/processor/producer/websocket/client/MessageWebSocketOverHttpSender.java

```diff
@@ -0,0 +1,182 @@
+        WebSocket wsClient = null;
+        try {
+            final SslEngineFactory ssl = getSslEngineFactory();
+
+            DefaultAsyncHttpClientConfig clientConfig = new DefaultAsyncHttpClientConfig.Builder()
+                    .setDisableHttpsEndpointIdentificationAlgorithm(true)
+                    .setUseOpenSsl(true)
+                    .setSslEngineFactory(ssl)
+                    .build();
+
+            WebSocketUpgradeHandler.Builder upgradeHandlerBuilder
+                    = new WebSocketUpgradeHandler.Builder();
+            WebSocketUpgradeHandler wsHandler = upgradeHandlerBuilder
+                    .addWebSocketListener(webSocketClientConfiguration.inputStreamSocketListenerWebSocketClient()).build();
+            wsClient = asyncHttpClient(clientConfig)
+                    .prepareGet(WS_URL)
+                    .execute(wsHandler)
+                    .get();
+            return wsClient;
+        } catch (Exception e) {
+            logger.info("... can not create the WebSocket connection");
+            if (null != message)
+                rejectionMessageServiceImpl.sendRejectionMessage(
+                        RejectionMessageType.REJECTION_COMMUNICATION_LOCAL_ISSUES,
+                        message);
+        }
+        return wsClient;
+    }
+
+    @NotNull
+    private SslEngineFactory getSslEngineFactory() throws NoSuchAlgorithmException, KeyManagementException {
+        final TrustManager[] trustAllCerts = new TrustManager[]{
+                new X509TrustManager() {
+                    @Override
+                    public void checkClientTrusted(java.security.cert.X509Certificate[] chain,
+                                                   String authType) throws CertificateException {
+                    }
+
+                    @Override
+                    public void checkServerTrusted(java.security.cert.X509Certificate[] chain,
+                                                   String authType) throws CertificateException {
+                    }
+
+                    @Override
+                    public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                        return new java.security.cert.X509Certificate[0];
+                    }
+                }
+        };
+        // Install the all-trusting trust manager
+        final SSLContext sslContext = SSLContext.getInstance("SSL");
+        sslContext.init(null, trustAllCerts, new java.security.SecureRandom());
+        return new JsseSslEngineFactory(sslContext);
+    }
+
+
+    private void closeWSClient(WebSocket wsClient, Message message) {
+        // Send the close frame 1000 (CLOSE), "Shutdown"; in this method we also close the wsClient.
+        try {
+            wsClient.sendCloseFrame(1000, "Shutdown");
+        } catch (Exception e) {
+            logger.error("Problems encountered during Client Shutdown with error: " + e.getMessage());
+            if (null != message)
```

Comment:
- In all situation when we throw the exception we should send the response message....

Reply:
I think the Stacktrace may be speaking enough!